app/controllers/v3/auth/federation_controller.rb

```diff
@@ -21,6 +25,16 @@ def auth_response(type, *filters)
         headers[x_subject_token_header_key] = Utils::Tokenator.to_token(credentials.to_hash)
         respond_with token_response
       end
+
+      def audit_unscoped_tokens
```

Comment:
Method does pretty much the same thing as method `audit_scoped_tokens` in `V3::Auth::TokensController`. Unite them. Preferably in some concern.